### PR TITLE
 kv,restore: log which value caused conversion failure

### PR DESCRIPTION
**Pull request by kennytm**
### What problem does this PR solve? 

1. When there is invalid data, the current log is not able to reveal what is the cause.
2. When a column is missing, Lightning will fill in default values for these columns, but this behavior may not be obvious.

### What is changed and how it works?

1. When KV encode failed, log the file path, row content, column index, name and type in additional to the error.
2. When a column is missing, log a warning about the missing column.

### Check List 

Tests 

 - Integration test

Side effects

Related changes

### CODE REVIEW

File: lightning/kv/sql2kv.go [hunk 1]
```diff
@@ -47,11 +50,76 @@ func (kvcodec *TableKVEncoder) Close() {
...
+			enc.AddInt("col", index)
+			enc.AddString("kind", kindStr[kind])
+			enc.AddString("val", str)
+			return nil
+		}))
+	}
+	return nil
+}
+
+func logKVConvertFailed(logger log.Logger, row []types.Datum, j int, colInfo *model.ColumnInfo, err error) {
```
Comment (GregoryIan) [inline lightning/kv/sql2kv.go]: Will not brush too much？

**Comment by GregoryIan** [comment]
Rest LGTM

File: lightning/kv/sql2kv.go (hunk 1 above)
Comment (kennytm) [reply lightning/kv/sql2kv.go]: What do you mean?

Comment (GregoryIan) [reply lightning/kv/sql2kv.go]: output too many logs

Comment (kennytm) [reply lightning/kv/sql2kv.go]: It will print at most a few per table with errors, since returning error should stop importing such tables.

**Comment by GregoryIan** [comment]
LGTM

**Pull request merged**